src/main/scala/mesosphere/marathon/api/v2/DeploymentsResource.scala

```diff
@@ -68,6 +68,10 @@ class DeploymentsResource @Inject() (
     currentStepInfo: DeploymentStepInfo): JsObject = {
 
     val steps = deployment.steps.map(step => step.actions.map(actionToMap)).map(Json.toJson(_))
+    val readinessResults = currentStepInfo.readinessChecksByApp.map{
+      case (id, results) => Json.obj(id.toString -> Json.obj("readinessCheckResults" -> Json.toJson(results)))
```

Comment:
you tested that, correct?

For me, this looks a bit incorrect. I think that you can actually include the currentStepInfo.readinessChecksByApp directly below -- without converting them first to JSON -- as long as the required formats are in scope.